dev-client/src/screens/SoilScreen/ColorScreen/ColorScreen.tsx

```diff
@@ -66,6 +71,16 @@ export const ColorScreen = (props: SoilPitInputScreenProps) => {
 
   const dispatch = useDispatch();
 
+  const userRole = useSelector(state =>
+    selectUserRoleSite(state, props.siteId),
+  );
+
+  const canDelete = useMemo(
+    () =>
+      userRole && ['MANAGER', 'CONTRIBUTOR', 'OWNER'].includes(userRole.role),
+    [userRole],
+  );
```

Comment:
`useSiteRoleContext()` couldn't be used here because site wasn't in scope. The site ID is in `props.`siteId. Would another approach be better?

Reply:
Could you extract this & the similar logic you've added in other places to a helper method somewhere in the `permissions` directory? I think it's fine if the site context isn't available but we can at least make this logic more centralized.